zebra-rpc/src/lib.rs

```diff
@@ -15,3 +15,82 @@ pub mod indexer;
 
 #[cfg(test)]
 mod tests;
+
+pub use methods::types::{
+    get_block_template::{
+        fetch_state_tip_and_local_time, generate_coinbase_and_roots,
+        proposal::proposal_block_from_template,
+    },
+    submit_block::SubmitBlockChannel,
+};
+
+/// Types and constants needed by clients of Zebra's RPC server
+//
+// Re-exports types to build the API. We want to flatten the module substructure
+// which is split mostly to keep file sizes small. Additionally, we re-export
+// types from other crates which are exposed in the API.
+//
+// TODO:
+// - Consider exporting functions here as well
+// - Expand documentation?
+// - Move `hex_data` and `trees` modules to/under the `types` module and qualify type refs in `methods`?
+pub mod client {
+    /// Types needed by clients of Zebra's RPC server
+    pub mod types {
```

Comment:
If we do this I think we can also flatten `types` and `constants` and put everything into `client`

Reply:
Fixed in https://github.com/ZcashFoundation/zebra/pull/9585/commits/087a59cd1e659a9f6f5dc909473bab505c0f5f8f.